common/oauthTokenManager.go

```diff
@@ -785,6 +803,124 @@ func jsonToTokenInfo(b []byte) (*OAuthTokenInfo, error) {
 
 // ====================================================================================
 
+var globalGRPCOAuthTokenLock = &sync.RWMutex{}
+
+// Pass in the RLocker to the sync.Cond so we're only requesting read locks, not write locks.
+var globalRPCOAuthTokenState = &GRPCOAuthToken{Mutex: sync.NewCond(globalGRPCOAuthTokenLock.RLocker())}
+
+func init() {
```

Comment:
Do we need this in init() or we can move it to a function which can be called on general init of azcopy.

Reply:
Is there somewhere better you'd suggest it be?